worker-document/src/main/java/com/hpe/caf/worker/document/tasks/DocumentTask.java

```diff
@@ -126,13 +126,27 @@ protected WorkerResponse createWorkerResponseImpl()
         // If the response message includes any scripts then it is in the v2 message format
         final int resultMessageVersion = (documentWorkerResult.scripts == null) ? 1 : 2;
 
+        //get the setting enable exception on failure
+        final Boolean enableExceptionOnFailure = application.getConfiguration().getEnableExceptionOnFailure();
+        if(enableExceptionOnFailure==true) {
+            if(ChangeLogFunctions.hasFailures(changes)){
+                // Create the WorkerResponse object with Error
+                return new WorkerResponse(outputQueue,
+                        TaskStatus.RESULT_EXCEPTION,
+                        data,
```

Comment:
Should the data not only contain details of the failure(s) rather than the entire document?